pkg/hintrunner/zero/zerohint_ec.go

```diff
@@ -74,6 +74,131 @@ func createEcNegateHinter(resolver hintReferenceResolver) (hinter.Hinter, error)
 	return newEcNegateHint(point), nil
 }
 
+// DivModeNSafeDivPlusOne performs a safe division of the result obtained from
+// the multiplication of `res` and `b` subtracted by `a`, by `N`. It then adds 1
+// to the final result to ensure safety and prevent division by zero errors.
+//
+// `DivModeNSafeDivPlusOne` assigns the result as `value` in the current scope.
+func newDivModNSafeDivPlusOneHint() hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "DivModNSafeDivPlusOne",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> value = k_plus_one = safe_div(res * b - a, N) + 1
+			valueBig := new(big.Int)
+
+			resBig, err := hinter.GetVariableAs[*big.Int](&ctx.ScopeManager, "res")
+			if err != nil {
+				return err
+			}
+
+			aBig, err := hinter.GetVariableAs[*big.Int](&ctx.ScopeManager, "a")
+			if err != nil {
+				return err
+			}
+
+			bBig, err := hinter.GetVariableAs[*big.Int](&ctx.ScopeManager, "b")
+			if err != nil {
+				return err
+			}
+
+			nBig, err := hinter.GetVariableAs[*big.Int](&ctx.ScopeManager, "N")
+			if err != nil {
+				return err
+			}
+
+			valueBig.Mul(resBig, bBig)
+			valueBig.Sub(valueBig, aBig)
+
+			newValueBig, err := secp_utils.SafeDiv(valueBig, nBig)
+			if err != nil {
+				return err
+			}
+
+			newValueBig.Add(&newValueBig, big.NewInt(1))
+			return ctx.ScopeManager.AssignVariables(map[string]any{"value": &newValueBig, "k_plus_one": &newValueBig})
```

Comment:
here, you should only add `value` in scope.

`k_plus_one` cairo variable is retrieved with `value` and there is no hint that uses it

Reply:
got it!

Reply:
done! ✅